### ISSUE TITLE
Autocomplete names should be based on online chatters list

### ISSUE THREAD

**Issue by sylouuu** [mccxiv/tc#341]
Hi, Thanks for your awesome work!

I've experienced an issue with the chatters list. I wanted to talk with someone with autocomplete "first letter" + "tab" but this feature is case sensitive.

Quick example:

* I want to talk with someone called "Azerty"
* The user is "azerty" is the chatters list
* I type "a" + "tab" but doesn't work since the real name starts with "A"
* I'm confused because I didn't know the real name case

So I suggest to show the display name instead of the username. I know the Twitch API gives you both.

Best

**Comment by scsddeputy** [comment]
The api endpoint that is used to obtain the list of chatters does not provide the display name, only the username which will always be lowercase unless twitch modifies that endpoint. That being said, using the TAB autocomplete should not be case sensitive. After testing, its working correctly for me. so if the users display name in chat is SF_Something, and i type sf and hit TAB, then it does complete to SF_Something. 

So there might be another issue happening here. What version and what OS are you operating on?

**Comment by sylouuu** [comment]
OK I'm using 8.1.0 on Windows 10.

This issue seems happen randomly, go to any channel, try to first letter + tab on first chatters. It doesn't work every time. You may have right it couldn't be related to case sensitive.

Best

**Comment by mccxiv** [comment]
Hey @sylouuu 

People are added to the autocomplete list only after they've spoken, and remain there even if they go offline. I think this explains both of your issues?

It is not case sensitive to a + tab and A + tab should work the same.

Should we change it to grab people from the current online list only? I've thought about that before since the current system can be confusing

**Comment by sylouuu** [comment]
Hi @mccxiv 

As a chatter, I expect the autocomplete to fully match the current channel chatters. :)

If you do so it will fix both issues you are right! 👍 

Best

**Comment by mccxiv** [comment]
I was going to release but I just don't think this is good enough. 

If someone joins the room and says hello, you want to be able to respond - but you can't since their name isn't in the viewer list yet. 

I think what we need is a combination of both. I will delay the release until I nail this feature

**Comment by sylouuu** [comment]
Alright, you may use `new Set()` instead of `{}`, it's an easy to deal with duplicates. :)

Best
